source/grid/grid_generator.cc

```diff
@@ -5927,6 +5927,206 @@ namespace GridGenerator
     tria.set_manifold(0, SphericalManifold<dim>(p));
   }
 
+  // To work around an internal clang-13 error we need to split up the
+  // individual hyper shell functions. This has the added bonus of making the
+  // control flow easier to follow - some hyper shell functions call others.
+  namespace internal
+  {
+    namespace
+    {
+      void
+      hyper_shell_6(Triangulation<3> &tria,
+                    const Point<3> &  p,
+                    const double      inner_radius,
+                    const double      outer_radius)
+      {
+        std::vector<Point<3>>    vertices;
+        std::vector<CellData<3>> cells;
+
+        const double irad = inner_radius / std::sqrt(3.0);
+        const double orad = outer_radius / std::sqrt(3.0);
+
+        // Corner points of the cube [-1,1]^3
+        static const std::array<Point<3>, 8> hexahedron = {{{-1, -1, -1}, //
+                                                            {+1, -1, -1}, //
+                                                            {-1, +1, -1}, //
+                                                            {+1, +1, -1}, //
+                                                            {-1, -1, +1}, //
+                                                            {+1, -1, +1}, //
+                                                            {-1, +1, +1}, //
+                                                            {+1, +1, +1}}};
+
+        // Start with the shell bounded by two nested cubes
+        for (unsigned int i = 0; i < 8; ++i)
+          vertices.push_back(p + hexahedron[i] * irad);
+        for (unsigned int i = 0; i < 8; ++i)
+          vertices.push_back(p + hexahedron[i] * orad);
+
+        const unsigned int n_cells                   = 6;
+        const int          cell_vertices[n_cells][8] = {
+          {8, 9, 10, 11, 0, 1, 2, 3},    // bottom
+          {9, 11, 1, 3, 13, 15, 5, 7},   // right
+          {12, 13, 4, 5, 14, 15, 6, 7},  // top
+          {8, 0, 10, 2, 12, 4, 14, 6},   // left
+          {8, 9, 0, 1, 12, 13, 4, 5},    // front
+          {10, 2, 11, 3, 14, 6, 15, 7}}; // back
+
+        cells.resize(n_cells, CellData<3>());
+
+        for (unsigned int i = 0; i < n_cells; ++i)
+          {
+            for (const unsigned int j : GeometryInfo<3>::vertex_indices())
+              cells[i].vertices[j] = cell_vertices[i][j];
+            cells[i].material_id = 0;
+          }
+
+        tria.create_triangulation(vertices, cells, SubCellData());
+      }
+
+      void
+      hyper_shell_12(Triangulation<3> &tria,
+                     const Point<3> &  p,
+                     const double      inner_radius,
+                     const double      outer_radius)
+      {
+        std::vector<Point<3>>    vertices;
+        std::vector<CellData<3>> cells;
+
+        const double irad = inner_radius / std::sqrt(3.0);
+        const double orad = outer_radius / std::sqrt(3.0);
```

Comment:
Here you divide again by `sqrt(3)`, which you have already done on the arguments passed in to this function.

Reply:
I fixed this and a problem with manifolds.